pyproject.toml

```diff
@@ -30,7 +30,7 @@ dependencies = [
     "flax",
     "jax",
     "jaxopt",
-    "jaxtyping",
+    "jaxtyping>0.2.31",
```

Comment:
Is this a breaking change? Or did we already implicitly depend on the version being above 0.2.31 and just didn't have a lower bound listed here?

Reply:
This PR explicitly requries the version to be above 0.2.31 (the functionality in this PR breaks on lower versions). I'm not sure what the implicit lower bound was before.

Reply:
In which case, it's probably safer to log this as a breaking change in the CHANGELOG.